File: src/workloads/lua/math_loop.lua
-- math_loop: tight numeric loop. Concession workload — JS JITs are
-- expected to win. Establishes the ceiling difference; doesn't change
-- the embedding-cost argument.

local N = PARAMS.inner_iters

function step()
    local s = 0.0
    for i = 1, N do
        s = s + math.sqrt(i) * 0.5
    end
    return s
end

File: src/workloads/ts/math_loop.ts
const N_math = PARAMS.inner_iters;

function step_math_loop(): number {
    let s = 0.0;
    for (let i = 1; i <= N_math; i++) {
        s = s + Math.sqrt(i) * 0.5;
    }
    return s;
}

(globalThis as any).step = step_math_loop;
